LaTeX/Inhalt/04_Inhalt/clean_architecture.tex

```diff
@@ -7,7 +7,23 @@ \subsection{Verschlüsselung}
+
 \section{Adapter}
 \subsection{Datenbank}
 Die Klasse \href{https://github.com/moorts/Morik/blob/main/src/adapters/database/DatabaseInterface.h}{\textit{DatabaseInterface}} implementiert die eigentliche Funktionalität in Form der SQL-Anweisungen und führt diese über eine konkrete Implementierung der \textit{AbstractSqlDatabase} in der Datenbank aus. Somit geht keine Funktionalität verloren, wenn das Plugin durch eine andere Datenbanktechnologie ausgetauscht wird. Die Umsetzung der Funktionalität als SQL-Anweisungen bedeutet jedoch, dass ein Austausch des Plugins nur ohne Weiteres möglich ist, wenn die neue Datenbank ebenfalls eine SQL-Datenbank ist. Handelt es sich bei der neuen Datenbank jedoch beispielsweise um eine noSQL-Datenbank, so muss die Funktionalität, also die Abfragen, angepasst werden. Der Klasse \textit{DatabaseInterface} wird im Konstruktor eine \textit{AbstractSqlDatabase} übergeben, was die Dependency Injection umsetzt. Zusätzlich erbt die Klasse \textit{DatabaseInterface} von der abstrakten Klasse \href{https://github.com/moorts/Morik/blob/main/src/application/AbstractDatabaseInterface.h}{\textit{AbstractDatabaseInterface}}, die sich im Kern befindet, sodass die Abhängigkeit vom Adapter auf den Kern zeigt. Dies setzt also die Dependency Inversion um und ermöglicht Klassen im Kern eine Abhängigkeit auf das AbstractDatabaseInterface zu haben statt auf eine konkrete Implementierung davon.\\
 Auf die Benutzung von Prepared Statements wurde innerhalb des Adapters verzichtet, da die Datenbank lokal ist und nur der Benutzer Befehle auf ihr ausführt. Würde die Datenbank über eine öffentliche Schnittstelle angesteuert werden, so wäre dies nicht zu vernachlässigen.
+
+\subsection{User Interface}
+
+Die Klasse \href{TODO:ENTER LINK AFTER MERGE}{\textit{UIDataHelper}} implementiert die Funktionalität, die ein Benutzer zur Verfügung hat. Namentlich handelt es sich dabei um das Anlegen und Bearbeiten von Passwörtern. Diese Klasse ist separat vom entsprechenden Plugin, damit die Funktionalität nicht abhängig ist, von spezifischen Plugins. Neue User Interfaces können also implementiert/hinzugefügt werden, ohne diese Funktionen erneut zu implementieren.
+
+\section{Applikationskern}
+
+Der Kern des Programmes, befindlich in \href{https://github.com/moorts/Morik/tree/main/src/application}{application} Ordner, ist zuständig für die Umsetzung der Domänenkonzepte und enthält zusätzlich Pure Fabrication Klassen (bspw. \href{https://github.com/moorts/Morik/blob/main/src/application/EntryFactory.h}{\textit{EntryFactory}}) und Interfaces, welche von den Plugins implementiert werden (siehe oben). Der Applikationskern hat keine externen Abhängigkeiten, durch konsequente Anwendung von Dependency-Inversion. 
```

Comment:
1. Satz: "befindlich in application Ordner" -> in dem / im
Die EntryFactory ist keine Pure Fabrication Klasse, da es bei DDD Factories gibt. Ich würde als Beispiel hier eher `InstanceManager` oder `DatabaseColumn` nennen
letzter Satz: "Anwendung von Dependency-Inversion" -> "Anwendung der Dependency-Inversion"